File: api-ios/TimeTreeAPI/Sources/Model/TAccessToken.swift
//
//  TAccessToken.swift
//  TimeTreeAPI
//
//  Created by EngarNet on 2020/05/02.
//  Copyright © 2020 EngarNet. All rights reserved.
//

import Foundation
import TimeTreeAPICommon

public struct TAccessToken {
    public init(
        accessToken: String,
        tokenType: String,
        scope: String,
        createdAt: Foundation.Date
    ) {
        self.accessToken = accessToken
        self.tokenType = tokenType
        self.scope = scope
        self.createdAt = createdAt
    }
    
    public let accessToken: String
    public let tokenType: String
    public let scope: String
    public let createdAt: Foundation.Date
}

extension TokenResponse {
    func toModel() -> TAccessToken {
        return TAccessToken(
            accessToken: self.accessToken,
            tokenType: self.tokenType,
            scope: self.scope,
            createdAt: Foundation.Date(timeIntervalSince1970: TimeInterval(self.createdAt))
        )
    }
}

File: api-android/src/main/java/com/engarnet/timetree/model/TAccessToken.kt
package com.engarnet.timetree.model

import com.engarnet.timetree.api.v1.api.authorization.response.TokenResponse
import java.util.*

data class TAccessToken(
    val accessToken: String,
    val tokenType: String,
    val scope: String,
    val createdAt: Date
)

internal fun TokenResponse.toModel(): TAccessToken {
    val response = this
    return TAccessToken(
        accessToken = response.accessToken,
        tokenType = response.tokenType,
        scope = response.scope,
        createdAt = Date().apply { time = response.createdAt.toLong() }
    )
}